src/modules/map/components/openlayers/utils/usePrint.composable.js

```diff
@@ -0,0 +1,99 @@
+import { computed, ref } from 'vue'
+import { useStore } from 'vuex'
+
+import { abortPrintJob, createPrintJob, waitForPrintJobCompletion } from '@/api/print.api.js'
+import { getGenerateQRCodeUrl } from '@/api/qrcode.api.js'
+import log from '@/utils/logging'
+
+const dispatcher = { dispatcher: 'usePrint.composable' }
+
+/** @enum */
+export const PrintStatus = {
+    IDLE: 'IDLE',
+    PRINTING: 'PRINTING',
+    FINISHED_SUCCESSFULLY: 'FINISHED_SUCCESSFULLY',
+    FINISHED_FAILED: 'FINISHED_FAILED',
+}
+
+/**
+ * Gathering of all the logic that will trigger and manage a print request to service-print3
+ *
+ * @param {Map} map
+ */
+export function usePrint(map) {
+    const currentJobReference = ref(null)
+    /** @type {PrintStatus} */
+    const printStatus = ref(PrintStatus.IDLE)
+
+    const store = useStore()
+    const printLayout = computed(() => store.state.print.selectedLayout)
+    const printScale = computed(() => store.state.print.selectedScale)
+    const attributions = computed(() =>
+        store.getters.visibleLayers
+            .map((layer) => layer.attributions)
+            .map((attribution) => attribution.name)
+            .filter((attribution, index, self) => self.indexOf(attribution) === index)
+    )
+    const layersWithLegends = computed(() =>
+        store.getters.visibleLayers.filter((layer) => layer.hasLegend)
+    )
+    const lang = computed(() => store.state.i18n.lang)
+    const projection = computed(() => store.state.map.projection)
```

Comment:
For these computed, should we just retrieve once when we call the function to prevent impact on performancce? e.g. only retrieve from store when we call `createPrintJob`?

Reply:
That might be a good idea yeah, I'll try and see if it works as intended